src/components/tag/tag.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { ReactNode, useMemo } from 'react';
+import { StyleProp, View, ViewStyle } from 'react-native';
+
+import { AppColor, TagType } from 'src/common/enums';
+import { CheckIcon, Text } from '..';
+import styles from './styles';
+
+type TagProps = {
+  children: ReactNode;
+  hasIcon?: boolean;
+  tagType: TagType;
+  style?: StyleProp<ViewStyle>;
+};
+
+const Tag: React.FC<TagProps> = ({
+  children,
```

Comment:
If we expect nothing but simple text to be rendered by tag we may go with some `text` prop instead of `children`